modules/javafx.graphics/src/main/java/javafx/application/Platform.java

```diff
@@ -432,4 +436,261 @@ public static ReadOnlyBooleanProperty accessibilityActiveProperty() {
+     *         <tr><td>{@code GTK.unfocused_borders}</td><td>{@link Color}</td></tr>
+     *         <tr><td>{@code GTK.warning_color}</td><td>{@link Color}</td></tr>
+     *         <tr><td>{@code GTK.error_color}</td><td>{@link Color}</td></tr>
+     *         <tr><td>{@code GTK.success_color}</td><td>{@link Color}</td></tr>
+     *         <tr></tr>
+     *     </tbody>
+     * </table>
+     *
+     * @since 22
+     */
+    public interface Preferences extends ObservableMap<String, Object> {
+
+        /**
+         * The platform color scheme, which specifies whether applications should prefer light text on
+         * dark backgrounds, or dark text on light backgrounds. The value of this property is derived
+         * from the perceptual brightness of {@link #backgroundColorProperty() backgroundColor} in relation
+         * to {@link #foregroundColorProperty() foregroundColor} and defaults to {@link ColorScheme#LIGHT}
+         * if the platform does not report color preferences.
+         *
+         * @return the {@code colorScheme} property
+         * @defaultValue {@link ColorScheme#LIGHT}
+         */
+        ReadOnlyObjectProperty<ColorScheme> colorSchemeProperty();
+
+        ColorScheme getColorScheme();
+
+        /**
+         * The color used for background regions.
+         * <p>
+         * If the platform does not report a background color, this property defaults to {@code Color.WHITE}.
+         *
+         * @return the {@code backgroundColor} property
+         * @defaultValue {@link Color#WHITE}
+         */
+        ReadOnlyObjectProperty<Color> backgroundColorProperty();
+
+        Color getBackgroundColor();
+
+        /**
+         * The color used for foreground elements like text.
+         * <p>
+         * If the platform does not report a foreground color, this property defaults to {@code Color.BLACK}.
+         *
+         * @return the {@code foregroundColor} property
+         * @defaultValue {@code Color.BLACK}
+         */
+        ReadOnlyObjectProperty<Color> foregroundColorProperty();
+
+        Color getForegroundColor();
+
+        /**
+         * The accent color.
+         * <p>
+         * If the platform does not report an accent color, this property defaults to vivid blue
+         * (corresponding to the hex color value {@code #157EFB}).
+         *
+         * @return the {@code accentColor} property
+         * @defaultValue {@code #157EFB}
+         */
+        ReadOnlyObjectProperty<Color> accentColorProperty();
+
+        Color getAccentColor();
+
+        /**
+         * Returns an optional {@code Integer} to which the specified key is mapped.
+         *
+         * @param key the key
+         * @throws NullPointerException if {@code key} is null
+         * @throws IllegalArgumentException if a mapping exists, but the key is not mapped to an {@code Integer}
+         * @return the optional {@code Integer} to which the key is mapped
```

Comment:
Alternative 2 (return empty value if the key maps to a different type) is attractive because it doesn't require additional type validation for mappings that are not present, and gets rid of the sometimes unexpected IAE.

However, there's a major downside: the preferences map is, first and foremost, a `Map`. The typed getters are just a convenience to make it easier to work with this map. When a native toolkit reports a key-value mapping, it will be included in the map, and its value can be unconditionally retrieved with `Map.get(String)`.

When we use a typed getter, it should return `Optional.empty()` exactly if `get(String)` would return `null`. Returning an empty value when the wrong typed getter is called (which is a bug) is unexpected and basically hides what would be the equivalent of a `ClassCastException`. This rules out this approach.

Alternative 1 (always throw IAE if we call the wrong typed getter, whether or not we have a mapping) is not unexpected, and does not violate the `Optional.empty() <=> get(String)==null` invariant. This alternative is a bit heavier on the implementation side, as we must now keep well-known lists of key-type mappings around, and more importantly, keep them in sync with the native toolkit.

I've implemented this alternative and updated the specification of the typed getters similar to this:
```java
/**
 * Returns an optional {@code Integer} to which the specified key is mapped.
 *
 * @param key the key
 * @throws NullPointerException if {@code key} is null
 * @throws IllegalArgumentException if the key is not mappable to an {@code Integer}
 * @return the optional {@code Integer} to which the key is mapped
 */
```